agent/provider/src/cli/whitelist.rs

```diff
@@ -0,0 +1,200 @@
+use std::collections::HashMap;
+
+use structopt::StructOpt;
+
+use ya_manifest_utils::{
+    matching::domain::{pattern_to_id, DomainPattern, DomainPatterns},
+    ArgMatch,
+};
+use ya_utils_cli::{CommandOutput, ResponseTable};
+
+use crate::cli::println_conditional;
+use crate::startup_config::ProviderConfig;
+
+#[derive(StructOpt, Clone, Debug)]
+#[structopt(rename_all = "kebab-case")]
+pub enum WhitelistConfig {
+    /// List domain whitelist patterns
+    List,
+    /// Add new domain whitelist patterns
+    Add(Add),
+    /// Remove domain whitelist patterns
+    Remove(Remove),
+}
+
+#[derive(StructOpt, Clone, Debug)]
+#[structopt(rename_all = "kebab-case")]
+pub struct Add {
+    /// Domain whitelist patterns
+    #[structopt(long, short)]
+    patterns: Vec<String>,
+
+    /// Domain whitelist patterns type
+    #[structopt(long = "type", short = "t", default_value = "strict")]
+    pattern_type: ArgMatch,
+}
+
+#[derive(StructOpt, Clone, Debug)]
+#[structopt(rename_all = "kebab-case")]
+pub struct Remove {
+    /// Domain whitelist pattern ids.
+    ids: Vec<String>,
+}
```

Comment:
Please add appropriate structopt `help="..."` attributes to explain the non-trivial commands and arguments (specifically `Add::{patterns, pattern_type}`), and their possible values.

Reply:
I completely forgot about 'help' messages, but to be fair we do not have many of them in other commands.
Suitable messages have been added to both `whitelist` and 'keystore' CLI commands.
Done.